docs/design/noc-root-cert-design.md

```diff
@@ -0,0 +1,75 @@
+# NOC Root Certificate Transactions Design
+
+## User Stories
+
+### 1. Add NOC Root Certificate
+A Vendor with DCL write privilege can submit a transaction to add a NOC Root certificate associated with their Vendor ID.
+
+### 2. Revoke NOC Root Certificate
+A Vendor with DCL write privilege can submit a transaction to revoke a NOC Root certificate associated with their Vendor ID. 
+
+## Transactions
+
+### 1. ADD_NOC_X509_ROOT_CERTIFICATE
+This transaction adds a NOC Root Certificate owned by the Vendor.
+
+- Who can send: Vendor account
+  - VID-scoped NOC Root Certificate: The `vid` field in the certificate's subject must be equal to the Vendor account's VID.
+- Validation:
+  - The provided certificate must be a root certificate:
+    - `Issuer` == `Subject`
+    - `Authority Key Identifier` == `Subject Key Identifier`
+  - No existing certificate with the same `<Certificate's Issuer>:<Certificate's Serial Number>` combination.
+  - If certificates with the same `<Certificate's Subject>:<Certificate's Subject Key ID>` combination already exist:
+    - The sender must match the owner of the existing certificates.
+  - The signature (self-signature) and expiration date must be valid.
+- Parameters:
+  - cert: `string` - The NOC Root Certificate, encoded in X.509v3 PEM format. Can be a PEM string or a file path.
+- State Changes:
+  - `pki/ApprovedCertificates/value/<Subject>/<SubjectKeyID>`
+  - `pki/NOCRootCertificates/value/<VID>`
+- CLI Command:
+  - `dcld tx pki add-noc-x509-root-cert --certificate=<string-or-path> --from=<account>`
+
+### 2. REVOKE_NOC_X509_ROOT_CERTIFICATE
```

Comment:
I think it's an open question how Revocation should work here.
- Should it be Remove rather than Revoke
- Should we keep a track of Removed/Revoked certificates (soft-delete VS hard-delete)